src/muxer/yamux/yamux_frame.cpp

```diff
@@ -5,103 +5,87 @@
 
 #include <libp2p/muxer/yamux/yamux_frame.hpp>
 
-namespace {
-  using ByteArray = libp2p::common::ByteArray;
-
-  ByteArray &putUint8(ByteArray &buffer, uint8_t number) {
-    buffer.push_back(number);
-    return buffer;
-  }
-
-  ByteArray &putUint16NetworkOrder(ByteArray &buffer, uint16_t number) {
-    buffer.push_back(static_cast<unsigned char &&>((number)&0xFFu));
-    buffer.push_back(static_cast<unsigned char &&>((number >> 8u) & 0xFFu));
-    return buffer;
-  }
-
-  ByteArray &putUint32NetworkOrder(ByteArray &buffer, uint32_t number) {
-    buffer.push_back(static_cast<unsigned char &&>((number)&0xFFu));
-    buffer.push_back(static_cast<unsigned char &&>((number >> 8u) & 0xFFu));
-    buffer.push_back(static_cast<unsigned char &&>((number >> 16u) & 0xFFu));
-    buffer.push_back(static_cast<unsigned char &&>((number >> 24u) & 0xFFu));
-    return buffer;
-  }
-}  // namespace
+#include <arpa/inet.h>
+#include <libp2p/common/byteutil.hpp>
 
 namespace libp2p::connection {
-  ByteArray YamuxFrame::frameBytes(uint8_t version, FrameType type, Flag flag,
-                                   uint32_t stream_id, uint32_t length,
-                                   gsl::span<const uint8_t> data) {
-    // TODO(akvinikym) PRE-118 15.04.19: refactor with NetworkOrderEncoder, when
-    // implemented
-    ByteArray buffer{};
-    putUint32NetworkOrder(
-        putUint32NetworkOrder(
-            putUint16NetworkOrder(
-                putUint8(putUint8(buffer, version), static_cast<uint8_t>(type)),
+  YamuxFrame::ByteArray YamuxFrame::frameBytes(uint8_t version, FrameType type,
+                                               Flag flag, uint32_t stream_id,
+                                               uint32_t length,
+                                               gsl::span<const uint8_t> data) {
+    ByteArray bytes;
+    bytes.reserve(kHeaderLength);  // minimum header size
+    common::putUint32BE(
```

Comment:
Maybe it's worth it to at least make using common; here to make this part shorter?